src/Analyzers/MSTest.Analyzers/UseNullableForIsNullAndIsNotNullAssertionsAnalyzer.cs

```diff
@@ -0,0 +1,69 @@
+using Microsoft.CodeAnalysis.Operations;
+
+using MSTest.Analyzers.Helpers;
+
+namespace MSTest.Analyzers;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+public sealed class UseNullableForIsNullAndIsNotNullAssertionsAnalyzer : DiagnosticAnalyzer
+{
+    private static readonly LocalizableResourceString Title = new(nameof(Resources.UseNullableForIsNullAndIsNotNullAssertionsAnalyzerTitle), Resources.ResourceManager, typeof(Resources));
+    private static readonly LocalizableResourceString Description = new(nameof(Resources.UseNullableForIsNullAndIsNotNullAssertionsAnalyzerDescription), Resources.ResourceManager, typeof(Resources));
+    private static readonly LocalizableResourceString MessageFormat = new(nameof(Resources.UseNullableForIsNullAndIsNotNullAssertionsAnalyzerMessageFormat), Resources.ResourceManager, typeof(Resources));
+
+    internal static readonly DiagnosticDescriptor UseNullableForIsNullAndIsNotNullAssertionsAnalyzerRule = DiagnosticDescriptorHelper.Create(
+        DiagnosticIds.UseNullableForIsNullAndIsNotNullAssertionsAnalyzerRuleId,
+        Title,
+        MessageFormat,
+        Description,
+        Category.Usage,
+        DiagnosticSeverity.Info,
+        isEnabledByDefault: true);
+
+    public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; }
+        = ImmutableArray.Create(UseNullableForIsNullAndIsNotNullAssertionsAnalyzerRule);
+
+    public override void Initialize(AnalysisContext context)
+    {
+        context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+        context.EnableConcurrentExecution();
+
+        context.RegisterCompilationStartAction(context =>
+        {
+            if (context.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.MicrosoftVisualStudioTestToolsUnitTestingAssert, out INamedTypeSymbol? assertSymbol)
+                && context.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemNullable, out INamedTypeSymbol? nullableSymbol))
+            {
+                context.RegisterOperationAction(context => AnalyzeOperation(context, assertSymbol, nullableSymbol), OperationKind.Invocation);
+            }
+        });
+    }
+
+    private static void AnalyzeOperation(OperationAnalysisContext context, INamedTypeSymbol assertSymbol, INamedTypeSymbol? nullableSymbol)
+    {
+        var invocationOperation = (IInvocationOperation)context.Operation;
+
+        if ((invocationOperation.TargetMethod.Name != "IsNull" && invocationOperation.TargetMethod.Name != "IsNotNull")
+            || !SymbolEqualityComparer.Default.Equals(assertSymbol, invocationOperation.TargetMethod.ContainingType))
+        {
+            return;
+        }
+
+        foreach (IArgumentOperation argument in invocationOperation.Arguments)
+        {
+            if (SymbolEqualityComparer.Default.Equals(argument.Type, nullableSymbol) && argument.Parameter?.Name == "value") // not working.
```

Comment:
Did it not work because nullableSymbol was null? See my first comment in that case.

Otherwise, I suspect that you are seeing a conversion operation to object, which would make that if statement never enter. If that is the case, you should probably ignore any implicit conversion operations and check the type of the "original" argument instead.